### DISCUSSION TITLE
Converte o DNS para o IP correspondente

### DISCUSSION THREAD

**Discussion by seadramdesenv** [Flutterando/forum#6] [Perguntas e Respostas] [answered]
Bom Dia,

Estou com um caso onde uso o pacote **internet_connection_checker** para ficar monitorando a conexão com o servidor.
Acontece que o servidor tem o IP Interno e o Externo onde o externo pode ser o dns ao invés do IP direto.
Sendo assim o pacote da erro quando recebe um dns, hoje ele só aceita numero de IP diretamente, e não estou encontrando uma forma no dart de resolver o DNS para poder passar o IP.

Se alguém souber como fazer agradeço.

**Comment by ewertonls** [comment, marked as answer]
A classe [InternetAddress](https://api.dart.dev/stable/2.18.0/dart-io/InternetAddress-class.html) do `dart:io` tem o metodo estatico [lookup](https://api.dart.dev/stable/2.18.0/dart-io/InternetAddress/lookup.html) que resolve dominios em IPs.

Exemplo `main.dart`:
```dart
import 'dart:io';

void main() async {
  final ips = await InternetAddress.lookup('google.com');
  
  print(ips);
}
```
```console
$ dart main.dart
[InternetAddress('142.250.78.238', IPv4), InternetAddress('2800:3f0:4004:802::200e', IPv6)]
```

Voce pode tentar fazer o lookup antes de mandar o ip para o `internet_connection_checker`

**Reply by paulobreim** [reply]
mas porque você precisa do IP? Não pode monitorar pelo domínio?

**Reply by seadramdesenv** [reply]
> mas porque você precisa do IP? Não pode monitorar pelo domínio?

Essa pacote não aceita, pelo domínio da erro de endereço inválido

**Reply by seadramdesenv** [reply]
> A classe [InternetAddress](https://api.dart.dev/stable/2.18.0/dart-io/InternetAddress-class.html) do `dart:io` tem o metodo estatico [lookup](https://api.dart.dev/stable/2.18.0/dart-io/InternetAddress/lookup.html) que resolve dominios em IPs.
> 
> Exemplo `main.dart`:
> 
> ```dart
> import 'dart:io';
> 
> void main() async {
>   final ips = await InternetAddress.lookup('google.com');
>   
>   print(ips);
> }
> ```
> 
> ```
> $ dart main.dart
> [InternetAddress('142.250.78.238', IPv4), InternetAddress('2800:3f0:4004:802::200e', IPv6)]
> ```
> 
> Voce pode tentar fazer o lookup antes de mandar o ip para o `internet_connection_checker`

Eu tentei algo similar e me trouxe um ip que não é o meu externo real, vou tentar desse jeito que mandou, obrigado
